src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -58,56 +58,90 @@ async function kdf(hash_algo, X, length, param) {
+ * @param  {String}                 fingerprint  Recipient fingerprint
+ * @returns {Promise<{V: BN, C: BN}>}            Returns public part of ephemeral key and encoded session key
+ * @async
+ */
+async function encrypt(oid, cipher_algo, hash_algo, m, Q, fingerprint) {
+  const key = await genPublicEphemeralKey(oid, cipher_algo, hash_algo, Q, fingerprint);
+  const C = aes_kw.wrap(key.Z, m.toString());
+  return {
+    V: key.V,
     C: C
   };
 }
 
 /**
- * Decrypt and unwrap the value derived from session key
+ * Generate ECDHE secret from private key and public part of ephemeral key
  *
  * @param  {module:type/oid}        oid          Elliptic curve object identifier
  * @param  {module:enums.symmetric} cipher_algo  Symmetric cipher to use
  * @param  {module:enums.hash}      hash_algo    Hash algorithm to use
  * @param  {BN}                     V            Public part of ephemeral key
- * @param  {Uint8Array}             C            Encrypted and wrapped value derived from session key
  * @param  {Uint8Array}             d            Recipient private key
  * @param  {String}                 fingerprint  Recipient fingerprint
- * @returns {Promise<Uint8Array>}                Value derived from session
+ * @returns {Promise<Uint8Array>}                Generated ephemeral secret
  * @async
  */
-async function decrypt(oid, cipher_algo, hash_algo, V, C, d, fingerprint) {
+async function genPrivateEphemeralKey(oid, cipher_algo, hash_algo, V, d, fingerprint) {
   const curve = new Curve(oid);
   const param = buildEcdhParam(enums.publicKey.ecdh, oid, cipher_algo, hash_algo, fingerprint);
   cipher_algo = enums.read(enums.symmetric, cipher_algo);
   V = curve.keyFromPublic(V);
   d = curve.keyFromPrivate(d);
   const S = d.derive(V);
-  const Z = await kdf(hash_algo, S, cipher[cipher_algo].keySize, param);
+  return kdf(hash_algo, S, cipher[cipher_algo].keySize, param);
+}
+
+/**
+ * Decrypt and unwrap the value derived from session key
+ *
+ * @param  {module:type/oid}        oid          Elliptic curve object identifier
+ * @param  {module:enums.symmetric} cipher_algo  Symmetric cipher to use
+ * @param  {module:enums.hash}      hash_algo    Hash algorithm to use
+ * @param  {BN}                     V            Public part of ephemeral key
+ * @param  {Uint8Array}             C            Encrypted and wrapped value derived from session key
+ * @param  {Uint8Array}             d            Recipient private key
+ * @param  {String}                 fingerprint  Recipient fingerprint
+ * @returns {Promise<Uint8Array>}                Value derived from session
+ * @async
+ */
+async function decrypt(oid, cipher_algo, hash_algo, V, C, d, fingerprint) {
+  const Z = await genPrivateEphemeralKey(oid, cipher_algo, hash_algo, V, d, fingerprint);
   return new BN(aes_kw.unwrap(Z, C));
 }
 
-export default { encrypt, decrypt };
+export default { encrypt, decrypt, genPublicEphemeralKey, genPrivateEphemeralKey };
```

Comment:
This is a bit a matter of opinion but since it's technically a public function it might be worth writing out `generate` in full, even though the names are already kinda long
```suggestion
export default { encrypt, decrypt, generatePublicEphemeralKey, generatePrivateEphemeralKey };
```

Reply:
The name is made following the style of genKeyPair in ./Curves